src/main/java/net/mcreator/ui/workspace/resources/ResourceFilterModel.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.workspace.resources;
+
+import net.mcreator.ui.workspace.WorkspacePanel;
+
+import javax.swing.*;
+import java.util.*;
+import java.util.function.Predicate;
+
+/**
+ * <p>A general filter model that is used inside a {@link AbstractResourcePanel}.
+ * It defines how a general resource panel's list of elements behaves.</p>
+ *
+ * @param <T>
+ */
+public class ResourceFilterModel<T> extends DefaultListModel<T> {
+	private final List<T> items;
+	private final List<T> filterItems;
+	private final WorkspacePanel workspacePanel;
+	private final Predicate<T> refilterItemsFilter;
+	private final Comparator<T> sortingCondition;
+
+	/**
+	 *
+	 * @param workspacePanel <p>The {@link WorkspacePanel} of the current workspace</p>
+	 * @param refilterItemsFilter <p>Defines which elements should be contained inside the filtered list of elements.</p>
+	 * @param sortingCondition <p>Defines how the filtered elements will be ordered.</p>
+	 */
+	public ResourceFilterModel(WorkspacePanel workspacePanel, Predicate<T> refilterItemsFilter,

```

Comment:
Sorting condition could be reduced to resourceNameProvider to  be more logical what it does, refilterItemsFilter could become filter method argument I would say